src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
It would be possible to do so, but the compiler creates different types of Symbol nodes and treats them differently, so we label them differently here. The nodes that are tagged `DeclarationExpression` are all created the same way and get the same implementing types; this enum reflects their treatment in the compiler.